src/webportal/src/app/user/fabric/userView/BatchVirtualClustersEditor.jsx

```diff
@@ -0,0 +1,144 @@
+//
+// THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
+// BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+import React, {useContext, useState} from 'react';
+import {Modal, FontClassNames, PrimaryButton, DefaultButton, Stack, StackItem, Dropdown, mergeStyles, getTheme} from 'office-ui-fabric-react';
+import PropTypes from 'prop-types';
+import c from 'classnames';
+import t from '../../../components/tachyons.scss';
+
+import {updateUserVcRequest} from '../conn';
+
+import Context from './Context';
+
+export default function BatchVirtualClustersEditor({isOpen = false, hide}) {
+  const {allVCs, showMessageBox, refreshAllUsers, getSelectedUsers} = useContext(Context);
+
+  const [virtualClusters, setVirtualClusters] = useState([]);
+
+  const handleVCsChanged = (_event, option, _index) => {
+    if (option.selected) {
+      virtualClusters.push(option.text);
+    } else {
+      virtualClusters.splice(virtualClusters.indexOf(option.text), 1);
+    }
+    setVirtualClusters(virtualClusters.slice());
+  };
+
+  const [lock, setLock] = useState(false);
+  const [needRefreshAllUsers, setNeedRefreshAllUsers] = useState(false);
+
+  const handleSubmit = async (event) => {
+    event.preventDefault();
+    setLock(true);
+
+    const virtualClustersString = virtualClusters.sort().join(',');
+    const users = getSelectedUsers();
+    for (let i = 0; i < users.length; i++) {
+      const user = users[i];
+      const result = await updateUserVcRequest(user.username, virtualClustersString)
+        .then(() => {
+          setNeedRefreshAllUsers(true);
+          return {success: true};
+        })
+        .catch((err) => {
+          return {success: false, message: String(err)};
+        });
+      if (!result.success) {
+        await showMessageBox(result.message);
+        setLock(false);
+        return;
+      }
+    }
+    await showMessageBox('Update vitrual clusters successfully');
+    setLock(false);
+    hide();
+    refreshAllUsers();
+  };
+
+  const handleCancel = () => {
+    hide();
+    if (needRefreshAllUsers) {
+      refreshAllUsers();
+    }
+  };
+
+  const tdPaddingStyle = c(t.pa3);
+  const tdLabelStyle = c(tdPaddingStyle, t.tr);
+
+  /**
+   * @type {import('office-ui-fabric-react').IDropdownOption[]}
+   */
+  const vcsOptions = allVCs.map((vc) => {
+    return {key: vc, text: vc};
+  });
+
+  const {spacing} = getTheme();
+
+  return (
+    <Modal
+      isOpen={isOpen}
+      containerClassName={mergeStyles({maxWidth: '430px'}, t.w90)}
```

Comment:
I think you'd better add `isBlocking` property if you want to disable the light dismiss feature

Reply:
Good suggestion, fixed.